Assets/Scripts/Player/PlayerController.cs

```diff
@@ -19,6 +24,7 @@ public class PlayerController : MonoBehaviour {
         rb2D = GetComponent<Rigidbody2D>();
         targetPosition = rb2D.position;
         currentHealth = maxHealth;
+        InvokeRepeating("HealthDecay", healthDecayTime, healthDecayTime);

```

Comment:
I think it's better for the automatic health decay to be smoother by setting both `healthDecayDamage` and `healthDecayTime` to 1.
I would like to use a float for `currentHealth` to make the decay even more smooth. Then, health could be decreased in the `Update` method using a health decay speed. You can make that change if you want to.